sdk/messaging_servicebus/src/utils.rs

```diff
@@ -0,0 +1,35 @@
+use azure_core::error::{Error, ErrorKind, ResultExt};
+use chrono::Duration;
+use url::Url;
+
+pub fn craft_peek_lock_url(
+    namespace: &str,
+    queue: &str,
+    lock_expiry: Option<Duration>,
+) -> Result<Url, Error> {
+    let mut url = Url::parse(&format!(
+        "https://{}.servicebus.windows.net/{}/messages/head",
+        namespace, queue
+    ))
+    .context(
+        ErrorKind::DataConversion,
+        "failed to parse peek_lock_message URL",
+    )?;
+
+    // add timeout, if given
+    if let Some(t) = lock_expiry {
+        url.query_pairs_mut()
+            .append_pair("timeout", &t.num_seconds().to_string());
+    };
+
+    Ok(url)
+}
+
+pub fn body_bytes_to_utf8(bytes: &[u8]) -> Result<String, Error> {
```

Comment:
I haven't looked at this library enough to know if a conversion to a `String` for a body is the right thing or if it should be passed around as `Bytes`.

Reply:
I followed what the crate initially did there. Plus, as a user of the crate, I'd want `String` rather than having to add a dependency for `Bytes` and handle that.

Reply:
We are okay with returning `Bytes`. A recent similar change I made is [here](https://github.com/Azure/azure-sdk-for-rust/pull/776/files#diff-5e3e9406014c7c90e2c7dd3f7616bc517d15bfdc63b60e47b55eb7bdd7d2a024L66). I'm not going to hold up this PR for this.